messaging/messaging.go

```
@@ -0,0 +1,454 @@
+	s := &struct {
+		Alert            interface{} `json:"alert,omitempty"`
+		ContentAvailable *int        `json:"content-available,omitempty"`
+		*apsAlias
+	}{
+		apsAlias: (*apsAlias)(a),
+	}
+
+	if a.Alert != nil {
+		s.Alert = a.Alert
+	} else {
+		s.Alert = a.AlertString
+	}
+	if a.ContentAvailable {
+		one := 1
+		s.ContentAvailable = &one
+	}
+	return json.Marshal(s)
+}
+
+// ApsAlert is the alert payload that can be included in an Aps.
+//
+// See https://developer.apple.com/library/content/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/PayloadKeyReference.html
+// for supported fields.
+type ApsAlert struct {
+	Title        string   `json:"title,omitempty"` // if specified, overrides the Title field of the Notification type
+	Body         string   `json:"body,omitempty"`  // if specified, overrides the Body field of the Notification type
+	LocKey       string   `json:"loc-key,omitempty"`
+	LocArgs      []string `json:"loc-args,omitempty"`
+	TitleLocKey  string   `json:"title-loc-key,omitempty"`
+	TitleLocArgs []string `json:"title-loc-args,omitempty"`
+	ActionLocKey string   `json:"action-loc-key,omitempty"`
+	LaunchImage  string   `json:"launch-image,omitempty"`
+}
+
+// ErrorInfo is a topic management error.
+type ErrorInfo struct {
+	Index  int
+	Reason string
+}
+
+// TopicManagementResponse is the result produced by topic management operations.
+//
+// TopicManagementResponse provides an overview of how many input tokens were successfully handled,
+// and how many failed. In case of failures, the Errors list provides specific details concerning
+// each error.
+type TopicManagementResponse struct {
+	SuccessCount int
+	FailureCount int
+	Errors       []*ErrorInfo
+}
+
+func newTopicManagementResponse(resp *iidResponse) *TopicManagementResponse {
+	tmr := &TopicManagementResponse{}
+	for idx, res := range resp.Results {
+		if len(res) == 0 {
+			tmr.SuccessCount++
+		} else {
+			tmr.FailureCount++
+			code, _ := res["error"].(string)
```

If there's a rationale for explicitly ignoring the error from the type assertion, let's document it. 

If the rationale is "// should never happen", we can just omit the _ and let it panic if a non-string is found.

---

I suppose if we've made this far into the error parsing process, we can assume the `error` key is present and contains a string value. Dropping the `_`.